R/atype.R

```diff
@@ -76,16 +80,24 @@ atype = function(
         mod = unique( valmap$mod ),
         stringsAsFactors = FALSE
       )
-      
-      # Check logical.
-      if( check_logical ){
+
+      # remove NAs
+      uvals_clean = uvals %>% purrr::keep(~!all(is.na(.))) 
+      uvals_clean = na.omit(uvals)
+
+      # for checking types, it'll be faster if we use less data.
+      uvals_sample = spl(uvals_clean, sample, warn = FALSE)
```

Comment:
A few key concepts here:
* Lines 86 actually overrides the work done by 85 so 85 is not necessary. 
* We should avoid using packages unnecessarily since this is likely to be slower and creates dependencies, which risks breaking our package if another package changes. keep comes from purr so we want to avoid that if we can. I do really like how you found na.omit, which is in stats:: which is part of the default R packages. So nice job there! I think you just forgot to remove 85?
* Avoid using variables only once, there is no need to define a variable if there is only one use. 

Please replace 85-89 with just this: ` uvals_sample = spl(na.omit(uvals), n = use_n_sampled_rows , warn = FALSE)`